mne/event.py

```diff
@@ -885,3 +886,354 @@ def concatenate_events(events, first_samps, last_samps):
+
+    @property
+    def _categories_in_use(self):
+        return {k: v for k, v in self._categories.items() if v['state']}
+
+    @property
+    def _events_in_use(self):
+        return {k: v for k, v in self._events.items() if v['in_use']}
+
+    def get_condition(self, raw, conditions=None, stim_channel=None, mask=None,
+                      uint_cast=None, mask_type='and'):
+        """ Get averaging parameters for a condition (averaging category).
+
+        Output is designed to be used with the Epochs class to extract the
+        corresponding epochs.
+
+        Parameters
+        ----------
+        raw : Raw object
+            An instance of Raw.
+        conditions : None | dict | list of dict
+            Condition or a list of conditions. Conditions can be strings
+            (DACQ comment field, e.g. 'Auditory left') or category dicts
+            (e.g. eav['Auditory left'], where eav is an instance of
+            ElektaAverager). If None, get all conditions marked active in
+            DACQ.
+        stim_channel : None | string | list of string
+            Name of the stim channel or all the stim channels
+            affected by the trigger. If None, the config variables
+            'MNE_STIM_CHANNEL', 'MNE_STIM_CHANNEL_1', 'MNE_STIM_CHANNEL_2',
+            etc. are read. If these are not found, it will fall back to
+            'STI101' or 'STI 014' if present, then fall back to the first
+            channel of type 'stim', if present.
+        mask : int | None
+            The value of the digital mask to apply to the stim channel values.
+            If None (default), no masking is performed.
+        uint_cast : bool
+            If True (default False), do a cast to ``uint16`` on the channel
+            data. This can be used to fix a bug with STI101 and STI014 in
+            Neuromag acquisition setups that use channel STI016 (channel 16
+            turns data into e.g. -32768), similar to ``mne_fix_stim14 --32``
+            in MNE-C.
+        mask_type: 'and' | 'not_and'
+            The type of operation between the mask and the trigger.
+            Choose 'and' for MNE-C masking behavior.
+
+        Returns
+        -------
+        conds_data : dict or list of dict, each with following keys:
+            events : array, shape (n_epochs_out, 3)
+                List of zero time points (t0) for the epochs matching the
+                condition. Use as the ``events`` parameter to Epochs. Note
+                that these are not (necessarily) actual events.
+            event_id : dict
+                Name of condition and index compatible with ``events``.
+                Should be passed as the ``event_id`` parameter to Epochs.
+            tmin : float
+                Epoch starting time relative to t0. Use as the ``tmin``
+                parameter to Epochs.
+            tmax : float
+                Epoch ending time relative to t0. Use as the ``tmax``
+                parameter to Epochs.
```

Comment:
@jaeilepp You mean the `event_id` returned by the class? That's always 1, since it's arbitrary (i.e. it just needs match the returned `events` array). The idea is that the calls to mne.Epochs will pick up one category at a time.

Now Epochs is multi-condition, so in principle it would be possible to return all categories once, i.e. interleave the events in a single array and give each one a unique id. The problem is that Epochs doesn't support different durations for different categories. Also (probably) the most common use case will be to do `epochs.average()`, which does not play well with multiple categories (since Evoked does not support them).